### PR TITLE
Correct docstrings for `HttpSensorAsync`

### LINKED ISSUE #706: Correct docstrings for `HttpSensorAsync`
**Describe the bug**
Update the docstrings for `HttpSensorAsync` as discussed in https://astronomer.slack.com/archives/C0387PMM2E7/p1665393838125069

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by rajaths010494**
Update the docstrings for HttpSensorAsync as discussed in https://astronomer.slack.com/archives/C0387PMM2E7/p1665393838125069..

image

closes #706

### CODE REVIEW

**Review by pankajastro — APPROVED** [review]

File: astronomer/providers/http/sensors/http.py [hunk 1]
```diff
@@ -42,7 +24,8 @@ def response_check(response, task_instance):
     :type request_params: a dictionary of string key/value pairs
     :param headers: The HTTP headers to be added to the GET request
     :type headers: a dictionary of string key/value pairs
-    :param response_check: A check against the 'requests' response object.
+    :param response_check: This param is currently not supported.
```
Comment (pankajkoti) [inline astronomer/providers/http/sensors/http.py:27]: Should we remove this param from the docstring here altogether since we're not supporting it?

Comment (bharanidharan14) [reply astronomer/providers/http/sensors/http.py:27]: Yes, agreed with @pankajkoti  can we remove this?

**Review by kaxil — CHANGES_REQUESTED** [review]
Why remove this? Removing docstring won't mean that the param isn't supported for Async

**Comment by rajaths010494** [comment]
> Why remove this? Removing docstring won't mean that the param isn't supported for Async.
> 
> Instead just add a `log.warn` line in the Operator's **init** or execute method to say it will use Synchronous operation since you passed `response_check`

Yes sure will add the log message.

**Review by kaxil — APPROVED** [review]

**Pull request merged**